### PR TITLE
add preference to auto-record session

### LINKED ISSUE #1240: Provide preferences switch to enable in-app session recording
Switch should default to 'on'

### PR DESCRIPTION
**Pull request by brentpicasso**
Resolves #1240

### CODE REVIEW

File: autosportlabs/racecapture/settings/prefs.py [hunk 1]
```diff
@@ -173,8 +173,9 @@ def set_config_defaults(self):
         self.config.setdefault('preferences', 'config_file_dir', default_user_files_dir)
         self.config.setdefault('preferences', 'firmware_dir', default_user_files_dir)
         self.config.setdefault('preferences', 'import_datalog_dir', default_user_files_dir)
-        self.config.setdefault('preferences', 'first_time_setup', True)
-        self.config.setdefault('preferences', 'send_telemetry', False)
```
Comment (ryandoherty) [inline autosportlabs/racecapture/settings/prefs.py]: Why the change from boolean to strings? Also, I think this should have gone into a separate PR, seems unrelated.

File: resource/settings/settings.json [hunk 2]
```diff
@@ -10,6 +10,14 @@
     },
     {
         "type": "bool",
+        "title": "Record Session",
+        "desc": "App will automatically record data while in Dashboard mode and will be available as a saved session for immediate analysis",
```
Comment (ryandoherty) [inline resource/settings/settings.json]: Suggested wording:

Log Data
Automatically record data in app while in Dashboard mode. Data will be available as a saved session in the Analysis screen.

File: autosportlabs/racecapture/settings/prefs.py (hunk 1 above)
Comment (brentpicasso) [inline autosportlabs/racecapture/settings/prefs.py]: Actually because booleans are incorrect for how preferences work. The value stored is actually a string; when I set it to True, the preferences screen did not interpret it correctly. True needs to be represented as '1'.  To be consistent I fixed the telemetry default as well.

File: resource/settings/settings.json (hunk 2 above)
Comment (brentpicasso) [inline resource/settings/settings.json]: How about: 

"Automatically record data in app while in Dashboard mode. Data will be available as a saved session in Analysis mode."

So we consistently use the word 'mode'

File: autosportlabs/racecapture/settings/prefs.py (hunk 1 above)
Comment (ryandoherty) [inline autosportlabs/racecapture/settings/prefs.py]: Ok. Huh, then how did this work before? Typecasting?

Comment (brentpicasso) [inline autosportlabs/racecapture/settings/prefs.py]: Yeah. it typecasted False to '0' but it won't typecast True to '1' . Since telemetry default was 'False' we never noticed it until now.
